HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -3071,120 +3163,239 @@ def pre_solve(self):
         self.update_solution_terminal()
         if not self.quiet:
             self.check_conditions(verbose=self.verbose)
+            
+            
+    def describe_parameters(self):
+        '''
+        Generate a string describing the primitive model parameters that will
+        be used to calculating limiting values and factors.
+        
+        Parameters
+        ----------
+        None
 
-    def check_GICNrm(self, verbose=None):
-        """
-        Check Individual Growth Patience Factor.
-        """
-        self.GPFNrm = self.thorn / (
-            self.PermGroFac[0] * self.InvEx_PermShkInv
-        )  # [url]/#GICRawI
+        Returns
+        -------
+        param_desc : str
+            Description of primitive parameters.
+        '''
+        # Get parameter description from the perfect foresight model
+        param_desc = PerfForesightConsumerType.describe_parameters(self)
+        
+        # Make a new entry for weierstrass-p (the weird formatting here is to
+        # make it easier to adapt into the style of the superclass if we add more
+        # parameter reports later)
+        this_entry = ['WorstPrb', 'probability of worst income shock realization', '℘', False]
+        try:
+            val = getattr(self,this_entry[0])
+        except:
+            val = self.bilt[this_entry[0]]    
+        this_line = this_entry[2] + f'={val:.5f} : ' + this_entry[1] + ' (' + this_entry[0] + ')\n'
```

Comment:
This way of constructing a string from these 'entries' is repeated often.
This can be functionalized out.